test/database/collection.go

```diff
@@ -0,0 +1,54 @@
+package database
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+)
+
+type collectionClient struct {
```

Comment:
wow, this file is a thorn in the side!  Should it go into go-cosmosdb?
either way, comment that its purpose is to be able to fake out PartitionKeyRanges, maybe call it fakeCollectionClient or something, return NotImplemented errors wherever possible to avoid accidental use

Reply:
done